test/fake_clock.h

```diff
@@ -66,6 +66,16 @@ class FakeClock {
   static std::mutex mutex_;
 };
 
+// Traits that allow basic_waitable_timer::async_wait() to work with a
+// FakeClock.
+template<typename Clock>
+struct WaitTraits {
```

Comment:
You shouldn't need this anymore — I think we got rid of this in #207. The specialization below should take care of it.